codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -77,6 +84,17 @@ public DefaultHttp2ConnectionDecoder(Http2Connection connection,
         connection.local().flowController().frameWriter(encoder.frameWriter());
     }
 
+    public DefaultHttp2ConnectionDecoder(Http2Connection connection,
+                                         Http2ConnectionEncoder encoder,
+                                         Http2FrameReader frameReader,
+                                         Http2Settings initialSettings,
+                                         Http2PromisedRequestVerifier requestVerifier) throws Http2Exception {
+        this(connection, encoder, frameReader, requestVerifier);
+        if (initialSettings != null) {
+            applyLocalSettings(initialSettings);
```

Comment:
I agree. We should wait until we apply the settings until we get the settings ACK (what was done prior to this PR).

Regarding the approaches to rectify the ambiguity in initial state why don't we start by clarifying interfaces which are used for settings and indicate the default expectations, and our default implementations only allow the default state to be set.

We can always implement the apply "default settings" approach on top of this if it isn't sufficient.

WDYT ... I will update this PR to reflect this in the mean time.